GitUI/CommitInfo/CommitInfo.cs

```diff
@@ -224,6 +234,66 @@ private int GetRevisionHeaderHeight()
             return _headerResize.Height;
         }
 
+        private async Task ReloadCommitInfoAsync()
+        {
+            _RevisionHeader.BackColor = ColorHelper.MakeColorDarker(BackColor);
+
+            showContainedInBranchesToolStripMenuItem.Checked = AppSettings.CommitInfoShowContainedInBranchesLocal;
+            showContainedInBranchesRemoteToolStripMenuItem.Checked = AppSettings.CommitInfoShowContainedInBranchesRemote;
+            showContainedInBranchesRemoteIfNoLocalToolStripMenuItem.Checked = AppSettings.CommitInfoShowContainedInBranchesRemoteIfNoLocal;
+            showContainedInTagsToolStripMenuItem.Checked = AppSettings.CommitInfoShowContainedInTags;
+            showMessagesOfAnnotatedTagsToolStripMenuItem.Checked = AppSettings.ShowAnnotatedTagsMessages;
+
+            ResetTextAndImage();
+
+            if (string.IsNullOrEmpty(_revision.Guid))
+            {
+                Debug.Fail("Unexpectedly called ReloadCommitInfoAsync() with empty revision");
+                return;
+            }
+
+            _RevisionHeader.Text = string.Empty;
+            _RevisionHeader.Refresh();
+
+            CommitData data = _commitDataManager.CreateFromRevision(_revision);
+            data.ChildrenGuids = _children;
+
+            if (_revision.Body == null)
+            {
+                // TODO error response here is ignored
+                await _commitDataManager.UpdateCommitMessageAsync(data, _revision.Guid);
+                _revision.Body = data.Body;
+            }
+
+            ThreadPool.QueueUserWorkItem(_ => loadLinksForRevision(_revision));
```

Comment:
❗️ Asynchronous operation not captured in the lifetime of the enclosing operation must be explicitly tracked

:bulb: Use tasks instead of `QueueUserWorkItem` (this occurs several times, but only this one is mentioned)